javascript/javascript3/week1/cityWeather.js

```diff
@@ -0,0 +1,155 @@
+
+                // Display the data
+                // This data should be showed in your app:
+
+                // The chosen city
+                // Temperature
+                // Icon for the weather type
+                // Wind speed
+                // How clowdy it is
+                // When sunrise and sunset is
+                // Optional a map showing where the city is located
+
+function weatherToMyChoice(){
+const button = document.getElementById('weatherButton')
+button.addEventListener('click', inputChange)
+
+const WeatherLocationButton = document.getElementById('locationButton')
+WeatherLocationButton.addEventListener('click', weatherLocationCapture)
+
+const saveLocation = document.getElementById('saveLocation')
+saveLocation.addEventListener('click',saveToLocalhost)
+
+const body = document.querySelector('body')
+const div = document.createElement('div')
+body.appendChild(div)
+
+let newCity = document.getElementById('cityname')
+newCity.addEventListener('change',inputChange);
+retrieveFromLocalhost( fetchTheWeather)
+
+
+//    input textbox changed && weather button callback
+function inputChange(){
+    if (newCity.value){
+    fetchTheWeather(newCity.value);
+}
+else {
+  newCity.value = 'insert valid city name'
+}}
+
+//  fetch weather by city name
+
+function fetchTheWeather(city){
+    fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=1fa178d7ac2b383598fae81ce7953bc7&units=metric`)
+    .then((response)=> response.json())
+    
+    .then((weather)=> {
+    
+        writeDataFromFetch(weather)
+    
+    })
+    .catch((error) => {
+      console.error('This is not a city!!! ', error);
+    })}
+//  fetch weathe by location
+function fetchFromLocation(Latitude,longitude){
+    fetch(`http://api.openweathermap.org/data/2.5/weather?lat=${Latitude}&lon=${longitude}&appid=1fa178d7ac2b383598fae81ce7953bc7&units=metric`)
+    .then((response)=> response.json())
+    
+    .then((weather)=> {
+    
+        writeDataFromFetch(weather);
+    })
+}
+
+ function writeDataFromFetch(weather){
+    newCity.value = weather.name;
+    div.innerHTML = `City : ${weather.name}`+"<br>"
+    div.innerHTML += `Temperature :  ${weather.main.temp}&#176c`+"<br>"
+    const weatherDesc = weather.weather[0].description
+    div.innerHTML += weatherDesc.charAt(0).toUpperCase()+weatherDesc.slice(1) + iconWriter(weatherDesc)+'<br>'
+    div.innerHTML += `Wind speed : ${weather.wind.speed}m/s`+"<br>"
+    div.innerHTML += `It is ${weather.clouds.all}% Coudy`+"<br>"
+    div.innerHTML += `Sunrise at : ${convertToTime(weather.sys.sunrise)}`+"<br>"
+    div.innerHTML += `Sunset at : ${convertToTime(weather.sys.sunset)}`+"<br>"
+
+ }
```

Comment:
Can find better name for function. 
suggestion:
standard nams like "renderData", "renderOutput" or can be more specific like "renderWeatherData" can be used for functions which render the data on browser.